From d78d99d3f58178703008be058d93016c41553d99 Mon Sep 17 00:00:00 2001
From: Friedemann Kleint <Friedemann.Kleint@qt.io>
Date: Tue, 9 Sep 2025 10:51:11 +0200
Subject: [PATCH 022/239] Gallery example: Brush up the .cpp file

Use modern string literals and replace #if-defery by a check on
QOperatingSystemVersion::currentType().

Pick-to: 6.9
Change-Id: I319b6c7665a4c4cadd7d1e711d8527f79ca35261
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>
(cherry picked from commit a6da6b3acaf6016893e66e5788aacc5adb9a117a)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 examples/quickcontrols/gallery/gallery.cpp | 54 ++++++++++++----------
 1 file changed, 30 insertions(+), 24 deletions(-)

diff --git a/examples/quickcontrols/gallery/gallery.cpp b/examples/quickcontrols/gallery/gallery.cpp
index b283748bc0..4cf66ccc97 100644
--- a/examples/quickcontrols/gallery/gallery.cpp
+++ b/examples/quickcontrols/gallery/gallery.cpp
@@ -1,51 +1,57 @@
 // Copyright (C) 2017 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR BSD-3-Clause
 
-#include <QGuiApplication>
+#include <QQuickStyle>
+
 #include <QQmlApplicationEngine>
 #include <QQmlContext>
-#include <QSettings>
-#include <QQuickStyle>
+
+#include <QGuiApplication>
 #include <QIcon>
 
+#include <QSettings>
+#include <QOperatingSystemVersion>
+
+using namespace Qt::StringLiterals;
+
+static constexpr auto styleKey = "style"_L1;
+
 int main(int argc, char *argv[])
 {
-    QGuiApplication::setApplicationName("Gallery");
-    QGuiApplication::setOrganizationName("QtProject");
+    QCoreApplication::setApplicationName("Gallery"_L1);
+    QCoreApplication::setOrganizationName("QtProject"_L1);
 
     QGuiApplication app(argc, argv);
 
-    QIcon::setThemeName("gallery");
+    QIcon::setThemeName("gallery"_L1);
 
     QSettings settings;
     if (qEnvironmentVariableIsEmpty("QT_QUICK_CONTROLS_STYLE"))
-        QQuickStyle::setStyle(settings.value("style").toString());
+        QQuickStyle::setStyle(settings.value(styleKey).toString());
 
     // If this is the first time we're running the application,
     // we need to set a style in the settings so that the QML
     // can find it in the list of built-in styles.
-    const QString styleInSettings = settings.value("style").toString();
+    const QString styleInSettings = settings.value(styleKey).toString();
     if (styleInSettings.isEmpty())
-        settings.setValue(QLatin1String("style"), QQuickStyle::name());
+        settings.setValue(styleKey, QQuickStyle::name());
 
     QQmlApplicationEngine engine;
 
-    QStringList builtInStyles = { QLatin1String("Basic"), QLatin1String("Fusion"),
-        QLatin1String("Imagine"), QLatin1String("Material"), QLatin1String("Universal"),
-        QLatin1String("FluentWinUI3") };
-#if defined(Q_OS_MACOS)
-    builtInStyles << QLatin1String("macOS");
-    builtInStyles << QLatin1String("iOS");
-#elif defined(Q_OS_IOS)
-    builtInStyles << QLatin1String("iOS");
-#elif defined(Q_OS_WINDOWS)
-    builtInStyles << QLatin1String("Windows");
-#endif
-
-    engine.setInitialProperties({{ "builtInStyles", builtInStyles }});
-    engine.load(QUrl("qrc:/gallery.qml"));
+    QStringList builtInStyles = { "Basic"_L1, "Fusion"_L1, "Imagine"_L1,
+        "Material"_L1, "Universal"_L1, "FluentWinUI3"_L1 };
+
+    if constexpr (QOperatingSystemVersion::currentType() == QOperatingSystemVersion::MacOS)
+        builtInStyles << "macOS"_L1 << "iOS"_L1;
+    else if constexpr (QOperatingSystemVersion::currentType() == QOperatingSystemVersion::IOS)
+        builtInStyles << "iOS"_L1;
+    else if constexpr (QOperatingSystemVersion::currentType() == QOperatingSystemVersion::Windows)
+        builtInStyles << "Windows"_L1;
+
+    engine.setInitialProperties({{ "builtInStyles"_L1, builtInStyles }});
+    engine.load(QUrl("qrc:/gallery.qml"_L1));
     if (engine.rootObjects().isEmpty())
         return -1;
 
-    return app.exec();
+    return QCoreApplication::exec();
 }
-- 
2.51.2

